src/Alert.php

```diff
@@ -0,0 +1,601 @@
+     *
+     * @param string $value
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return static
+     */
+    public function headerTag(string $value): self
+    {
+        if (empty($value)) {
+            throw new InvalidArgumentException('Header tag must be a string and cannot be empty.');
+        }
+
+        $new = clone $this;
+        $new->headerTag = $value;
+        return $new;
+    }
+
+    /**
+     * The attributes for rendering the `<i>` tag for icons alerts.
+     *
+     * @param array $value
+     *
+     * @return static
+     *
+     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     */
+    public function iconAttributes(array $value): self
+    {
+        $new = clone $this;
+        $new->iconAttributes = $value;
+        return $new;
+    }
+
+    /**
+     * Set icon css class in the alert component.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function iconClass(string $value): self
+    {
+        $new = clone $this;
+        Html::addCssClass($new->iconAttributes, $value);
+        return $new;
+    }
+
+    /**
+     * The attributes for rendering the container `<i>` tag.
+     *
+     * The rest of the options will be rendered as the HTML attributes of the `<i>` tag.
+     *
+     * @param array $value
+     *
+     * @return static
+     *
+     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     */
+    public function iconContainerAttributes(array $value): self
+    {
+        $new = clone $this;
+        $new->iconContainerAttributes = $value;
+        return $new;
+    }
+
+    /**
+     * The CSS class for the container `<i>` tag.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function iconContainerClass(string $value): self
+    {
+        $new = clone $this;
+        Html::addCssClass($new->iconContainerAttributes, $value);
+        return $new;
+    }
+
+    /**
+     * Set icon text in the alert component.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function iconText(string $value): self
+    {
+        $new = clone $this;
+        $new->iconText = $value;
+        return $new;
+    }
+
+    /**
+     * Set layout the alert panel body.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function layoutBody(string $value): self
+    {
+        $new = clone $this;
+        $new->layoutBody = $value;
+        return $new;
+    }
+
+    /**
+     * Set layout the alert panel header.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function layoutHeader(string $value): self
+    {
+        $new = clone $this;
+        $new->layoutHeader = $value;
+        return $new;
+    }
+
+    protected function run(): string
+    {
+        return $this->renderAlert();
```

Comment:
`renderAlert` method isn't needed. Code could be placed right in the `run()`method.

Reply:
done.